internal/osv/osv.go

```diff
@@ -119,6 +121,26 @@ func checkResponseError(resp *http.Response) error {
 	return fmt.Errorf("server response error: %s", string(respBuf))
 }
 
+func makeRetryRequest(method string, url string, data io.Reader) (*http.Response, error) {
+	var resp *http.Response
+	var err error
+	retries := 3
+	for i := 0; i < retries; i++ {
+		if method == "GET" {
+			resp, err = http.Get(url)
+		} else if method == "POST" {
+			resp, err = http.Post(url, "application/json", data)
+		} else {
+			return nil, fmt.Errorf("Invalid HTTP method: %s", method)
+		}
+		if err == nil {
+			break
+		}
+		time.Sleep(time.Second)
+	}
+	return resp, err
+}
+
```

Comment:
Lets make this a bit more general and typo proof.
```suggestion
func makeRetryRequest(action func() (*http.Response, error)) (*http.Response, error) {
	var resp *http.Response
	var err error
	retries := 3
	for i := 0; i < retries; i++ {
		resp, err = action()
		if err == nil {
			break
		}
		time.Sleep(time.Second)
	}
	return resp, err
}
```

Now the calls can pass a function in to do whatever request is required.